### PR TITLE
Add a `Motivation` section in README to explain the problem solved by CLP IR.

### PR DESCRIPTION
**Pull request by LinZhihao-723**
# Description
As `clp-ffi-py` is developed, there are more details we can discuss in terms of CLP IR. This PR clarifies the problem CLP IR solved and the benefit of using our logging library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,28 @@
 # CLP Python Logging Library
 
 This is a Python `logging` library meant to supplement [CLP (Compressed Log Processor)][0].
-Logs are compressed in a streaming fashion into CLP's Internal Representation (IR) format before written to disk.
-More details are described in this [Uber's blog][1].
+It operates by serializing and compressing log events using the CLP Intermediate Representation (IR)
+format, achieving both data streaming capability and effective compression ratios. Log files
```
Comment (davidlion) [inline README.md]:
```suggestion
format, achieving both data streaming capabilities and effective compression ratios. Log files
```

File: README.md [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+serialized using IR format can be viewed using [YScope Log Viewer][2]. They can also be deserialized
```
Comment (davidlion) [inline README.md]:
```suggestion
serialized using the IR format can be viewed using the [YScope Log Viewer][2]. They can also be deserialized
```

**Review by davidlion — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -93,6 +97,11 @@ CLPSockHandler(Path("example.clp.zst")).stop_listener()
 
 ## CLP readers (decoders)
 
+Important Update: The readers and all the other non-logging APIs currently available in this library
```
Comment (kirkrodrigues) [inline README.md]: Can use GitHub markdown [alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).

File: README.md [hunk 4]
```diff
@@ -93,6 +97,11 @@ CLPSockHandler(Path("example.clp.zst")).stop_listener()
...
+are scheduled for deprecation in the upcoming release. To access our newest and improved CLP IR
+analytic interface, which offers advanced features like high-performance decoding and enhanced query
+search capabilities, please check and install [clp-ffi-py][9].
```
Comment (kirkrodrigues) [inline README.md]:
```suggestion
are scheduled for deprecation in an upcoming release. To access our newest and improved CLP IR
analytics interface (which offers advanced features like high-performance decoding and enhanced query
search capabilities) check out [clp-ffi-py][9].
```

File: README.md [hunk 5]
```diff
@@ -256,3 +265,4 @@ word][7].
 [6]: https://black.readthedocs.io/en/stable/index.html
 [7]: https://docformatter.readthedocs.io/en/latest/faq.html#interaction-with-black
 [8]: https://beta.ruff.rs/docs/
+[9]: https://github.com/y-scope/clp-ffi-py
```
Comment (kirkrodrigues) [inline README.md]: Finish files with empty lines.

File: README.md [hunk 6]
```diff
@@ -1,24 +1,28 @@
...
+efficient querying through standard text-based tools like `grep`.
 
-This logging library helps solve this problem by logging directly in CLP's
-Internal Representation (IR). A log created with a CLP logging handler is first
-parsed and then appended to a compressed output stream in IR form.
-See [README-protocol.md](README-protocol.md) for more details on the format of
-CLP IR.
+To address this problem, this logging library is designed to serialize log events directly in CLP's
+Intermediate Representation (IR) format. A log event created with a CLP logging handler will be
+first encoded into the IR format, and then appended to a compressed output stream. This approach not
```
Comment (kirkrodrigues) [inline README.md]:
```suggestion
Intermediate Representation (IR) format. A log event created with a CLP logging handler will first
be encoded into the IR format, and then appended to a compressed output stream. This approach not
```

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -93,6 +97,12 @@ CLPSockHandler(Path("example.clp.zst")).stop_listener()
 
 ## CLP readers (decoders)
 
+> [!WARNING]
+> Important Update: The readers and all the other non-logging APIs currently available in this
+> library are scheduled for deprecation in an upcoming release. To access our newest and improved
+> CLP IR analytics interface (which offers advanced features like high-performance decoding and
+> enhanced query search capabilities) check out [clp-ffi-py][9].
```
Comment (kirkrodrigues) [inline README.md]:
```suggestion
> The readers and all the other non-logging APIs currently available in this library are scheduled
> for deprecation in an upcoming release. To access our newest and improved CLP IR analytics
> interface (which offers advanced features like high-performance decoding and enhanced query search
> capabilities) check out [clp-ffi-py][9].
```

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

**Review by kirkrodrigues — APPROVED** [review]

**Review by davidlion — APPROVED** [review]

**Pull request merged**